### PR TITLE
Prepare UI for new Device statuses types

### PR DESCRIPTION
**Pull request by celdrake**
Since we'll soon have the three type of devices statuses available (Device Status, Application Status and Update Status), this PR prepares the UI to handle them.

- The new types have been generated from https://github.com/flightctl/flightctl/pull/298. We'll need to generate them from `main` later, but there shouldn't be many differences.

- The filter toolbar is prepared to show each status category and allow to select the desired statuses.

- The filters will work like the "fleeId" filter does, meaning they can be activated via the URL which will allow us to create links to the device list page with selected filters.

- By resetting the last commit, we can keep the existing UI until the API is ready. Then we can apply back those changes to get the new behaviour. Example video:

https://github.com/flightctl/flightctl-ui/assets/829045/710bb29b-ce52-4fbf-8b4b-4b9060682b06

### CODE REVIEW

File: libs/ui-components/src/components/Device/useDeviceFilters.ts [hunk 1]
```diff
@@ -1,46 +1,83 @@
...
 
-export const useDeviceFilters = (
-  resources: Array<Device | EnrollmentRequest>,
-  queryFilters: { filterByFleetId: string | undefined },
-) => {
-  const [filters, setFilters] = React.useState<{ status: Array<DeviceConditionStatus | ApprovalStatus> }>({
-    status: [],
-  });
-  const [fleetName, setFleetName] = React.useState<string | undefined>(queryFilters.filterByFleetId);
+const validDeviceStatuses = Object.values(DeviceSystemSummaryStatusType) as string[];
```
Comment (rawagner) [inline libs/ui-components/src/components/Device/useDeviceFilters.ts:20]: we are missing `Pending` status in the status dropdown

File: libs/ui-components/src/utils/status/devices.ts [hunk 2]
```diff
@@ -1,29 +1,112 @@
...
-    }
-  });
-  allUniqueStatuses.sort((a, b) => a.label.localeCompare(b.label));
-  return allUniqueStatuses;
-};
+export enum FilterSearchParams {
+  Fleet = 'fleetId',
+  Device = 'devSt',
+  App = 'appSt',
+  Update = 'updSt',
```
Comment (rawagner) [inline libs/ui-components/src/utils/status/devices.ts:17]:
```suggestion
  Fleet = 'fleetId',
  DeviceStatus = 'devSt',
  AppStatus = 'appSt',
  UpdateStatus = 'updSt',
```

**Review by rawagner — COMMENTED** [review]
A few nits, lets fix them in followup

**Review by rawagner — APPROVED** [review]

File: libs/ui-components/src/components/Device/useDeviceFilters.ts (hunk 1 above)
Comment (celdrake) [reply libs/ui-components/src/components/Device/useDeviceFilters.ts:20]: Yes, I'm handling this now in the new branch adapting from the new API types

File: libs/ui-components/src/utils/status/devices.ts (hunk 2 above)
Comment (celdrake) [reply libs/ui-components/src/utils/status/devices.ts:17]: I'll rename in the new branch

**Pull request merged**